src/OpenLoco/Windows/ObjectSelectionWindow.cpp

```diff
@@ -1,17 +1,1459 @@
+                dpi,
+                widgets[widx::objectImage].mid_x() + 1 + self->x,
+                widgets[widx::objectImage].mid_y() + 1 + self->y,
+                _50D15C);
+        }
+
+        auto x = self->widgets[widx::objectImage].mid_x() + self->x;
+        auto y = self->widgets[widx::objectImage].bottom + 3 + self->y;
+        auto width = self->width - self->widgets[widx::scrollview].right - 6;
+
+        {
+            auto buffer = const_cast<char*>(stringmgr::getString(string_ids::buffer_2039));
+
+            *buffer++ = control_codes::window_colour_2;
+            auto objectPtr = self->object;
+
+            strncpy(buffer, objectmgr::object_index_entry::read(&objectPtr)._name, 512);
+
+            gfx::drawStringCentredClipped(*dpi, x, y, width, colour::black, string_ids::buffer_2039);
+        }
+
+        {
+            auto objectPtr = self->object;
+
+            drawDescription(
+                objectmgr::object_index_entry::read(&objectPtr)._header,
+                self,
+                dpi,
+                self->widgets[widx::scrollview].right + self->x + 4,
+                y + descriptionRowHeight,
+                _50D15C);
+        }
+    }
+
+    // 0x0047361D
+    static void drawScroll(window* self, gfx::drawpixelinfo_t* dpi, uint32_t)
+    {
+        gfx::clearSingle(*dpi, colour::getShade(self->colours[1], 4));
+
+        if (objectmgr::getNumInstalledObjects() == 0)
+            return;
+
+        int y = 0;
+        auto objects = objectmgr::getAvailableObjects(static_cast<object_type>(self->current_tab));
+        for (auto [i, object] : objects)
+        {
+            uint8_t flags = (1 << 7) | (1 << 6) | (1 << 5);
+            gfx::fillRectInset(dpi, 2, y, 11, y + 10, self->colours[1], flags);
+
+            uint8_t textColour = control_codes::colour_black;
+
+            auto objectPtr = self->object;
+            if (objectPtr != nullptr)
+            {
+                auto windowObjectName = objectmgr::object_index_entry::read(&objectPtr)._name;
+                if (object._name == windowObjectName)
+                {
+                    gfx::fillRect(dpi, 0, y, self->width, y + rowHeight - 1, (1 << 25) | palette_index::index_30);
+                    textColour = control_codes::window_colour_2;
+                }
+            }
+
+            if (_50D144[i] & (1 << 0))
+            {
+                auto x = 2;
+                _currentFontSpriteBase = font::m2;
+
+                if (textColour != control_codes::window_colour_2)
+                {
+                    _currentFontSpriteBase = font::m1;
+                }
+
+                auto checkColour = colour::opaque(self->colours[1]);
+
+                if (_50D144[i] & 0x1C)
+                {
+                    checkColour = colour::inset(checkColour);
+                }
+
+                gfx::drawString(dpi, x, y, checkColour, _strCheckmark);
+            }
+
+            _byte_112CC04[0] = textColour;
```

Comment:
112CC04 is just any old buffer. Create your own local version of this.
`char buffer[512];`...